### ISSUE TITLE
Question about the saved models

### ISSUE THREAD

**Issue by viet2411** [KevinMusgrave/pytorch-metric-learning#563]
Hello,
Thank you for your project.

I have followed your example **TrainWithClassifier.ipynb** to train the model with Trainers, but there are too many saved files.

![image](https://user-images.githubusercontent.com/39814151/209813903-72312c85-f4f9-47d2-81d0-6b7322f489c5.png)

I don't really understand the differences between them. Can you explain it to me?
And how can I change the setting of saving files?

Thank you and best regards.

**Comment by KevinMusgrave** [comment]
The number "1" represents the epoch at which the files were saved. 

Files with "best" represent the checkpoint with the best accuracy.

"trunk" and "embedder" are the components of your model.

"metric_loss" and "tuple_miner" are your loss function and miner, if they have any parameters.

The trainers in this library are a bit limited. If you want more customizability, you can use this library's loss functions with training frameworks like PyTorch Lightning, PyTorch Ignite, and PyTorch Catalyst.
